pkg/idpool/idpool_test.go

```diff
@@ -74,240 +64,240 @@ func (s *IDPoolTestSuite) TestInsertRemoveIDs(c *C) {
 
 	// Removing leased IDs should make them unavailable.
 	p3 := poolWithAllIDsLeased()
 	for i := minID; i <= maxID; i++ {
-		c.Assert(p3.Remove(ID(i)), Equals, false)
+		require.Equal(t, false, p3.Remove(ID(i)))
 	}
 	// The IDs should not be leased anymore.
 	for i := minID; i <= maxID; i++ {
-		c.Assert(p3.Use(ID(i)), Equals, false)
+		require.Equal(t, false, p3.Use(ID(i)))
 	}
 	// They should be unavailable.
-	leaseAllIDs(p3, minID, minID-1, c)
+	leaseAllIDs(p3, minID, minID-1, t)
 }
 
-func (s *IDPoolTestSuite) TestOperationsOnUnavailableIDs(c *C) {
+func TestOperationsOnUnavailableIDs(t *testing.T) {
 	minID, maxID := 1, 5
 	var poolWithAllIDsUnavailable = func() *IDPool {
 		p := NewIDPool(ID(minID), ID(maxID))
 		for i := minID; i <= maxID; i++ {
-			c.Assert(p.Remove(ID(i)), Equals, true)
+			require.Equal(t, true, p.Remove(ID(i)))
 		}
 		return p
 	}
 
 	// Releasing unavailable IDs should not have any effect.
 	p1 := poolWithAllIDsUnavailable()
 	for i := minID; i <= maxID; i++ {
-		c.Assert(p1.Release(ID(i)), Equals, false)
+		require.Equal(t, false, p1.Release(ID(i)))
 	}
-	leaseAllIDs(p1, minID, minID-1, c)
+	leaseAllIDs(p1, minID, minID-1, t)
 
 	// Using unavailable IDs should not have any effect.
 	p2 := poolWithAllIDsUnavailable()
 	for i := minID; i <= maxID; i++ {
-		c.Assert(p2.Use(ID(i)), Equals, false)
+		require.Equal(t, false, p2.Use(ID(i)))
 	}
-	leaseAllIDs(p2, minID, minID-1, c)
+	leaseAllIDs(p2, minID, minID-1, t)
 
 	// Inserting unavailable IDs should make them available.
 	p3 := poolWithAllIDsUnavailable()
 	for i := minID; i <= maxID; i++ {
-		c.Assert(p3.Insert(ID(i)), Equals, true)
+		require.Equal(t, true, p3.Insert(ID(i)))
 	}
 	// They should not be leased.
 	for i := minID; i <= maxID; i++ {
-		c.Assert(p3.Use(ID(i)), Equals, false)
-		c.Assert(p3.Release(ID(i)), Equals, false)
+		require.Equal(t, false, p3.Use(ID(i)))
+		require.Equal(t, false, p3.Release(ID(i)))
 	}
-	leaseAllIDs(p3, minID, maxID, c)
+	leaseAllIDs(p3, minID, maxID, t)
 
 	// Removing unavailable IDs should not have any effect.
 	p4 := poolWithAllIDsUnavailable()
 	for i := minID; i <= maxID; i++ {
-		c.Assert(p4.Remove(ID(i)), Equals, false)
+		require.Equal(t, false, p4.Remove(ID(i)))
 	}
-	leaseAllIDs(p4, minID, minID-1, c)
+	leaseAllIDs(p4, minID, minID-1, t)
 }
 
-func leaseAllIDs(p *IDPool, minID int, maxID int, c *C) {
+func leaseAllIDs(p *IDPool, minID int, maxID int, t *testing.T) {
 	expected := make([]int, 0)
 	actual := make([]int, 0)
 	for i := minID; i <= maxID; i++ {
 		id := p.LeaseAvailableID()
-		c.Assert(id, Not(Equals), NoID)
+		require.NotEqual(t, NoID, id)
 		actual = append(actual, int(id))
 		expected = append(expected, i)
 	}
 	// We should be out of IDs.
 	id := p.LeaseAvailableID()
-	c.Assert(id, Equals, NoID)
+	require.Equal(t, NoID, id)
 
 	// Unique ids must have been leased.
 	sort.Ints(actual)
-	c.Assert(actual, checker.DeepEquals, expected)
+	require.EqualValues(t, actual, expected)
```

Comment:
`require.ElementsMatch` is more idiomatic (then you can also drop the sort operation above).

Reply:
ah good point :100: